### PR TITLE
boostification of simple http server

### PR DESCRIPTION
**Pull request by sima-fastly**
simplehttpserver's signals updated from qt signals to boost. 
all the corresponding connections were also updated.

Tested locally

### CODE REVIEW

File: src/cpp/statsmanager.cpp [hunk 1]
```diff
@@ -1556,8 +1556,8 @@ private slots:
 			).arg(prometheusPrefix, m.name, m.help, prometheusPrefix, m.name, m.type, prometheusPrefix, m.name, value.toString());
 		}
 
-		connect(req, &SimpleHttpRequest::finished, req, &QObject::deleteLater);
-
+		req->finished.connect(boost::bind(&SimpleHttpRequest::deleteLater, req));
+
```
Comment (jkarneges) [inline src/cpp/statsmanager.cpp:1560]: whitespace

File: src/cpp/simplehttpserver.cpp [hunk 2]
```diff
@@ -486,15 +487,15 @@ class SimpleHttpServerPrivate : public QObject
...
+private:
 	void server_newConnection()
 	{
 		if(local)
 		{
 			QLocalSocket *sock = ((QLocalServer *)server)->nextPendingConnection();
 			SimpleHttpRequest *req = new SimpleHttpRequest(maxHeadersSize, maxBodySize);
 			connect(req->d, &SimpleHttpRequest::Private::ready, this, &SimpleHttpServerPrivate::req_ready);
-			connect(req, &SimpleHttpRequest::finished, this, &SimpleHttpServerPrivate::req_finished);
+			finishedConnection = req->finished.connect(boost::bind(&SimpleHttpServerPrivate::req_finished, this, req));
```
Comment (jkarneges) [inline src/cpp/simplehttpserver.cpp:498]: We need to keep track of a connection per request here, or else every new request overwrites the one connection.

File: src/cpp/simplehttpserver.cpp [hunk 3]
```diff
@@ -553,7 +553,11 @@ SimpleHttpRequest *SimpleHttpServer::takeNext()
 	if(!d->pending.isEmpty())
 	{
 		SimpleHttpRequest *req = d->pending.takeFirst();
-		req->disconnect(d);
+		auto conn = d->finishedConnections.find(req);
+		if (conn != d->finishedConnections.end()) {
+			conn->second.disconnect();
```
Comment (jkarneges) [inline src/cpp/simplehttpserver.cpp:558]: Since these are scoped connections, maybe deleting the entry is enough and this call is not needed?

**Review by jkarneges — APPROVED** [review]

File: src/cpp/simplehttpserver.cpp (hunk 3 above)
Comment (sima-fastly) [reply src/cpp/simplehttpserver.cpp]: very true! done

File: src/cpp/simplehttpserver.cpp [hunk 4]
```diff
@@ -553,7 +553,10 @@ SimpleHttpRequest *SimpleHttpServer::takeNext()
 	if(!d->pending.isEmpty())
 	{
 		SimpleHttpRequest *req = d->pending.takeFirst();
-		req->disconnect(d);
+		auto conn = d->finishedConnections.find(req);
+		if (conn != d->finishedConnections.end())
+			d->finishedConnections.erase(conn);
```
Comment (jkarneges) [inline src/cpp/simplehttpserver.cpp:558]: You may even be able to write this more simply as: `d->finishedConnections.erase(req)`.

**Review by jkarneges — APPROVED** [review]

**Pull request merged**